### DISCUSSION TITLE
R Shiny Workarounds for database.db.tmp Folder in Read-Only

### DISCUSSION THREAD

**Discussion by danoobae** [duckdb/duckdb#2967] [Q&A] [answered]
Is there a way to do either of the following:
Option 1: Prevent the need for `database.db.tmp` being created/utilized ([reference issue #58](https://github.com/duckdb/duckdb/issues/58#issuecomment-583026454))

Option 2: Prevent the `database.db.tmp` folder from being removed after connection close. Perhaps just have all the contents in the folder removed.

Context: I have an R Shiny app deployed using RStudio Server and Shiny Server, both in docker containers with read-only permission. The app will not function when the `database.db.tmp` folder is not present, and this folder cannot be created on application start-up because of the permission level. When I manually create the `database.db.tmp` folder in RStudio Server, the app works, but I have come across frequent instances when the folder is automatically removed. I am still unsure how it is able to delete the folder with read-only permission to the container. One known trigger that causes the folder to be deleted is if the application crashes.

Any insight or workaround would be much appreciated!

**Comment by Mytherin** [comment, marked as answer]
The `database.db.tmp` folder should not be created or required unless the memory threshold of the machine is exceeded. One way of making sure this never happens is to set the memory limit to infinite (`pragma memory_limit=-1`), in which case offloading to disk should never happen and the tmp directory should never be created.

**Reply by danoobae** [reply]
Awesome, thanks for the info!
